src/bmi_pet.c

```diff
@@ -772,7 +773,7 @@ int read_init_config_pet(pet_model* model, const char* config_file)//,
         }
 
     } // end loop through config
-
+    fclose(fp);
```

Comment:
While it is unlikely, `fclose` is not guaranteed to succeed. Both instances of this should be changed to something like:

```suggestion
    if (fclose(fp)){
        perror("fclose");
    }
```

Reply:
While I agree that failure is exceedingly unlikely, if it occurs, it should absolutely be checked and reported

In that vein, the call in `read_file_line_counts_pet` should be checked as well.

Reply:
Just want to put this one back on the radar. Are we still waiting on the proposed changes per @aaraney & @PhilMiller ? -thx

Reply:
There is really nothing you can do if `fclose` fails -- at least not in a generally portable way .

```
Whether or not the call succeeds, the stream shall be disassociated from the file and any buffer set by the setbuf() or setvbuf() function shall be disassociated from the stream. If the associated buffer was automatically allocated, it shall be deallocated.
```

If there is an issue/error, its at the OS level and the best you can do is read the errno and report it, but you can't recover or handle errors, and this isn't nearly as relevant when opening in 'r' mode either, because you don't need to worry about data integrity in the file.

So if we get to this point and there is an error in closing the file, but we have properly read it to this point and initialized things from it, I don't think warrants a `BMI_FAILURE` -- we should be safe to continue.  Worst case, we end up with a potential resource leak which lead us here in the first place.

It is theoretically possible, I think, that stream is disassociated without the FD being returned to the OS.  However, if the sys calls ad OS are in that bad of shape, it probably doesn't really matter  what we do here, not going to make much difference. If every FD gets leaked, then we are back to where we started, but now its the OS fault, not this code for not trying.

That said, we can report an `errorno` here if we really feel like it brings any value, but I'm not convinced it does.

Reply:
I agree that it is very unlikely to occur _and_ nothing actionable can be done if it does occur. I'll leave it to @PhilMiller to have the final say.

Reply:
For the config file, yeah, that reasoning makes sense. If the system is broken, we'll find out soon enough later in the run, or everything else will work.